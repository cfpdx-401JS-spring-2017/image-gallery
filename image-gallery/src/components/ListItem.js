import React, { Component } from 'react';

class ListItem extends Component {
  

  render() {
   let {images}= this.props;
   if (!images.length) return <div> loading... </div>

   return(
     <div className='ListItem'>
      <h2>{images.title}</h2>
     </div>
    );
  }
}

export default ListItem;
