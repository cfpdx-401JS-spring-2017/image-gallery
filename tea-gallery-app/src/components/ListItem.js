import React from 'react';
import PropTypes from 'prop-types';
import { DeleteImageButton } from './DeleteButtons';

ListItem.PropTypes = {
  images: PropTypes.array.required,
  onDeleteImage: PropTypes.func.required
};

export default function ListItem(props) {

  return (
    <ul>
      {props.images.map(image => (
        <div className="list-item-wrapper" key={image.title}>
          <li className="image-list-item">Title: {image.title}</li>
          <li className="image-list-item">Description: {image.description}</li>
          <li className="image-list-item"><a href={image.url}>Image</a></li>
          <DeleteImageButton
            images={props.images}
            onDeleteImage={() => props.onDeleteImage(image._id)}
          />
        </div>
      ))}
      {props.images.length < 1 && <h2>Album is empty. Add images below</h2>}
    </ul>
  );
}