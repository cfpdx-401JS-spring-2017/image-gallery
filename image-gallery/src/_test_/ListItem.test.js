import React from 'react';
import renderer from 'react-test-renderer';
import ListItem from '../components/ListItem';

const images = [

  {
    title: 'Riot Grrrl 1',
    description: 'Kathleen Hanna',
    url: 'http://www.newyorker.com/wp-content/uploads/2012/11/121126_r22838_g2048-1200.jpg'
  },

  {
    title: 'Riot Grrrl 2',
    description: 'Corrin Tucker',
    url: 'https://s-media-cache-ak0.pinimg.com/originals/a1/1a/d9/a11ad9fb2cf4c687fcb1212db5da342c.jpg'
  },

  {
    title: 'Riot Grrrl 3',
    description: 'Every Grrrl',
    url: 'https://www.speakertv.com/wp-content/uploads/2015/01/every-girl-is-a-riot-girl.jpg'
  }
];


describe('ListItem', () => {
  it('lists images as items', () => {
    const tree = renderer.create(<ListItem images={images} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});