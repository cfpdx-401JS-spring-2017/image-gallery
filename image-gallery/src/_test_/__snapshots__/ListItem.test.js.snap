// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem lists images as items 1`] = `
<div
  className="ListItem"
>
  <ul>
    <li>
      Riot Grrrl 1
    </li>
    <li>
      Kathleen Hanna
    </li>
    <li>
      http://www.newyorker.com/wp-content/uploads/2012/11/121126_r22838_g2048-1200.jpg
    </li>
  </ul>
  <ul>
    <li>
      Riot Grrrl 2
    </li>
    <li>
      Corrin Tucker
    </li>
    <li>
      https://s-media-cache-ak0.pinimg.com/originals/a1/1a/d9/a11ad9fb2cf4c687fcb1212db5da342c.jpg
    </li>
  </ul>
  <ul>
    <li>
      Riot Grrrl 3
    </li>
    <li>
      Every Grrrl
    </li>
    <li>
      https://www.speakertv.com/wp-content/uploads/2015/01/every-girl-is-a-riot-girl.jpg
    </li>
  </ul>
</div>
`;
